Hashtable.c

```diff
@@ -109,13 +109,15 @@ static size_t nextPrime(size_t n) {
 }
 
 Hashtable* Hashtable_new(size_t size, bool owner) {
-   Hashtable* this;
-
-   this = xMalloc(sizeof(Hashtable));
-   this->items = 0;
-   this->size = size ? nextPrime(size) : 13;
-   this->buckets = (HashtableItem*) xCalloc(this->size, sizeof(HashtableItem));
-   this->owner = owner;
+   size = size ? nextPrime(size) : 13;
+
+   Hashtable* this = xMalloc(sizeof(Hashtable));
+   *this = (Hashtable) {
+      .items = 0,
+      .size = size,
+      .buckets = (HashtableItem*) xCalloc(size, sizeof(HashtableItem)),
```

Comment:
I personally don't like this writing style, especially for object initialization code.

It might look ok for htop codebase (since htop doesn't check for allocation failures and just crashes if that happens), but for other projects, it is a better practice to allocate member buffers first, before the main object buffer. That way it would be easier to rewind the process (freeing the members) in case of a failure.